cpp/arcticdb/version/version_core.cpp

```diff
@@ -525,6 +536,10 @@ void check_column_and_date_range_filterable(const pipelines::index::IndexSegment
     "The data for this symbol is pickled and does not support date_range, row_range, or column queries");
     util::check(index_segment_reader.has_timestamp_index() || !std::holds_alternative<IndexRange>(read_query.row_filter),
             "Cannot apply date range filter to symbol with non-timestamp index");
+    sorting::check<ErrorCode::E_UNSORTED_DATA>(index_segment_reader.tsd().stream_descriptor().sorted() == arcticc::pb2::descriptors_pb2::SortedValue::UNKNOWN ||
+        index_segment_reader.tsd().stream_descriptor().sorted() == arcticc::pb2::descriptors_pb2::SortedValue::ASCENDING ||
+        !std::holds_alternative<IndexRange>(read_query.row_filter),
```

Comment:
The index type check is already preformed above.

You might also want to extract `index_segment_reader.tsd().stream_descriptor().sorted()` out to a variable.

Reply:
Certainly, it could be factored out and save some extra check. However, the impact of this saving would be marginal compared, at expenses of losing clarity in the code to understand what exactly we are checking here, which I think it is  important. On the other hand, we might want to change the definition of some of the checks in the future, and it could become a problem.